lib/utils.js

```diff
@@ -54,11 +56,27 @@ var utils = module.exports = {
    * Return the data argument from a list of arguments
    */
   getDataFromArgs: function(args) {
-    if (args.length > 0) {
-      if (isPlainObject(args[0]) && !utils.isOptionsHash(args[0])) {
-        return args.shift();
-      }
+    if (args.length < 1 || !isPlainObject(args[0])) {
+      return {};
     }
+
+    if (!utils.isOptionsHash(args[0])) {
+      return args.shift();
+    }
+
+    var argKeys = Object.keys(args[0]);
+
+    var optionKeysInArgs = argKeys.filter(function(key) {
+      return OPTIONS_KEYS.indexOf(key) > -1;
+    });
+
+    if (argKeys.length > optionKeysInArgs.length) {
```

Comment:
Wouldn't this conditional be a little more intuitive if we just compare `optionKeysInargs.length > 0`?

Reply:
Yes.  Yes it would. 😂